src/main.rs

```diff
@@ -302,10 +315,18 @@ fn run() -> Result<()> {
     curl.http_version(HttpVersion::Any)?;
     curl.http_09_allowed(true)?;
 
-    if opt.verbose {
+    if opt.verbose || opt.debug {
         curl.verbose(true)?;
     }
 
+    if opt.debug {
+        curl.debug_function(|info, data| {
+            eprintln!("\n-----\nDEBUG - infotype = {:?}\n-----", info);
+            let _ = io::stderr().write_all(data);
+            eprintln!("-----");
```

Comment:
I would get rid of this `"-----"` at the end of the debug block due to binary data messing up the ASCII output. The initial new line characters are enough to create a separation. I removed it on pings as well